src/procedures/InvestInTieredSto.ts

```diff
@@ -15,23 +15,44 @@ import { SecurityToken, TieredSto } from '../entities';
 import { ApproveErc20 } from './ApproveErc20';
 import { Factories } from '../Context';
 
+/**
+ * @hidden
+ */
 export const createRefreshSecurityTokenFactoryResolver = (
   factories: Factories,
   securityTokenId: string
 ) => async () => {
   return factories.securityTokenFactory.refresh(securityTokenId);
 };
 
+/**
+ * @hidden
+ */
 export const createRefreshTieredStoFactoryResolver = (
   factories: Factories,
   tieredStoId: string
 ) => async () => {
   return factories.tieredStoFactory.refresh(tieredStoId);
 };
 
+/**
+ * Procedure that invests in a Tiered STO
+ */
 export class InvestInTieredSto extends Procedure<InvestInTieredStoProcedureArgs> {
   public type = ProcedureType.InvestInTieredSto;
 
+  /**
+   * Invest the specified amount in the STO
+   *
+   * Note that this procedure will fail if:
+   * - The security token symbol doesn't exist
+   * - The STO address is invalid
+   * - The STO is either archived or hasn't been launched
+   * - The STO hasn't started yet
+   * - The STO is paused
+   * - The STO doesn't allow beneficial investments when trying to invest on behalf of someone else if the STO doesn't allow beneficial investments
```

Comment:
`- Trying to invest on behalf of someone else if the STO doesn't allow beneficial investments`

Reply:
This hasn't been addressed